src/parser/mod.rs

```diff
@@ -2352,12 +2352,14 @@ impl<'a> Parser<'a> {
     /// [map]: https://duckdb.org/docs/sql/data_types/map.html#creating-maps
     fn parse_duckdb_map_literal(&mut self) -> Result<Expr, ParserError> {
         self.expect_token(&Token::LBrace)?;
-
-        let fields = self.parse_comma_separated(Self::parse_duckdb_map_field)?;
-
-        self.expect_token(&Token::RBrace)?;
-
-        Ok(Expr::Map(Map { entries: fields }))
+        if self.peek_token().token == Token::RBrace {
```

Comment:
is it possible to switch to `let fields = self.parse_comma_separated0(Self::parse_duckdb_map_field)?;` instead? if so that would let us skip the if/else here

Reply:
I don't think we can do it. It seems that `parse_comma_separated0` is for parentheses, but MAP uses braces.
https://github.com/sqlparser-rs/sqlparser-rs/blob/d49acc67b13e1d68f2e6a25546161a68e165da4f/src/parser/mod.rs#L3487

Reply:
Ah i see, I'm thinking we could still reuse the logic (thinking its currently a bit odd that `parse_comma_separated0` is hardcoded to parenthesis while `parse_comma_separated` is token agnostic as it should be). Seems there's ony a couple of usages of `parse_comma_separated0`, we could change it to something like `parse_comma_separated0(|| {}, Token::RParen)`?

Reply:
I addressed this in 0763f52 and 5d81654. I share the same logic for `brace`, `parenthesis`, and `bracket`. I think it looks better. Thanks.